src/passes/StackIR.cpp

```diff
@@ -65,7 +69,39 @@ class StackIROptimizer {
     if (passOptions.optimizeLevel >= 3 || passOptions.shrinkLevel >= 1) {
       local2Stack();
     }
-    removeUnneededBlocks();
+    // Removing unneeded blocks is dangerous with GC, as if we do this:
+    //
+    //   (call
+    //     (rtt)
+    //     (block
+    //       (nop)
+    //       (i32)
+    //     )
+    //   )
+    // === remove inner block ==>
+    //   (call
+    //     (rtt)
+    //     (nop)
+    //     (i32)
+    //   )
+    //
+    // Then we end up with a nop that forces us to emit this during load:
+    //
+    //   (call
+    //     (block
+    //       (local.set
+    //         (rtt)
+    //       )
+    //       (nop)
+    //       (local.get)
+    //     )
+    //     (i32)
+    //   )
+    //
+    // However, that is not valid as an rtt cannot be set to a local.
```

Comment:
Even without GC, is it beneficial to remove blocks that have stacky code, or multiple return values anyway, if we have to introduce `local.set` and `local.get`?

Reply:
It is harmful if we roundtrip, yes. But it does emit a slightly smaller wasm in the middle, which can be shipped. It is a little awkward that binaryen has different behavior on stacky code than VMs. I wonder if we can fix that somehow, but I doubt it.